docs/ops/deployment/kubernetes.md

```diff
@@ -112,7 +112,7 @@ An early version of a [Flink Helm chart](https://github.com/docker-flink/example
 
 ### Session cluster resource definitions
 
-The Deployment definitions use the pre-built image `flink:latest` which can be found [on Docker Hub](https://hub.docker.com/r/_/flink/).
+The Deployment definitions use the pre-built image `flink:{{ site.version }}-scala_2.11` (or `flink:{{ site.version }}-scala_2.12`) which can be found [on Docker Hub](https://hub.docker.com/r/_/flink/).
```

Comment:
I am afraid we could not use `{{ site.version }}` here. We have at lease two reasons.
* When the minor version increase, the `{{ site.version }}` do not change. For example, 1.9.2 has released, and the `{{ site.version }}` is still 1.9.0.
* We do not have a image version for the master branch document.

Reply:
- Strange. I see here that the site.version for 1.9.2 was actually set to 1.9.2 : https://github.com/apache/flink/blob/release-1.9.2/docs/_config.yml#L30 . I do agree that for example https://ci.apache.org/projects/flink/flink-docs-release-1.9/dev/connectors/pubsub.html still shows 1.9.0. To me this seems more like the site needs to be regenerated than this variable having the incorrect value.

- Perhaps on releasing a 'SNAPSHOT' version of the software a 'SNAPSHOT' version docker image with the same software should be released too?

On the other hand ... it is 'just documentation'.
I do see that apparently the 2.12 is 'preferred' so I'm swapping them in the documentation.

Reply:
Releasing a `SNAPSHOT` image tag is still in discussion. Currently, Flink only provide the image for released versions.

Reply:
As I mentioned this change is not strictly related to the JIRA at hand, as it is about standalone kubernetes deployment and not about the native kubernetes image option.
It seems this `site.version` is not reliable at the moment. I agree with @wangyang0918,
this can bring a confusion.
I still think we should create another issue for this doc change which depends on 2 problems to be resolved: 
- the image Flink version should point to the latest minor version,
- there are no Flink snapshot versions in docker hub at the moment so master doc will point to a not existing image.

Reply:
We had the following idea with @zentol (you can check it in #12131):
```
flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %}
```

This way we keep it a concrete version for the stable released docs, like e.g. `1.10.1` (the bug fix release guide should include incrementing minor version in `site.version`, this should change after `1.11`). There is always a docker image for the released versions.

The master docs (not released, like 1.11-SNAPHOST) will always point to the `latest` released docker image for consistency for now until we introduce snapshots for docker image as well.

I know that the implicit scala version is currently `2.12` for short docker tags but it is `2.11` for everything else. `site.scala_version` is also `2.11`. It should have probably been `2.11` for docker tags as well.